src/Drips.sol

```diff
@@ -439,12 +439,14 @@ library Drips {
         uint32 cycleSecs = cycleSecs_;
         uint256 currIdx = 0;
         uint256 newIdx = 0;
-        while (true) {
-            bool pickCurr = currIdx < currReceivers.length;
+        bool pickCurr;
+        bool pickNew;
+        do {
```

Comment:
```suggestion
        bool pickCurr = true;
        bool pickNew = true;
        while(pickCurr || pickNew) {
```
This makes the context of the loop a little less stretched.

Personally I don't have a problem with `while(true)` loops, they aren't an antipattern, are they? IMO `pickCurr` and `pickNew` are meaningless and confusing outside of the context of an iteration which they describe, it suggests that their values can live across iterations.

Reply:
I agree that it is definitely not an anti-pattern. I think if the while condition is very complex it can be a lot of times more elegant to use a while(true) together with a break condition.

However, I think here the while condition is not that complex. 

In the end it is taste, here